pkg/types/telemetrytypes/field.go

```diff
@@ -175,11 +175,15 @@ func DataTypeCollisionHandledFieldName(key *TelemetryFieldKey, value any, tblFie
 		// is used which prepares the
 		// final query as attributes_number['http.status_code'] = 200 giving this error
 		// This following is one way to workaround it
+
+		// if the key is a number, the value is a string, we will let clickHouse handle the conversion
 		case float32, float64:
 			tblFieldName = castFloatHack(tblFieldName)
 		case string:
-			// try to convert the number attribute to string
-			tblFieldName = castString(tblFieldName) // numeric col vs string literal
+			if !isComparisonOperator {
+				// try to convert the number attribute to string
+				tblFieldName = castString(tblFieldName) // numeric col vs string literal
+			}
```

Comment:
We should also check if the `value` is indeed a number and if it's not a number we should cast it? For instance, if somebody does `severity_number > 'ERROR'` should we 5xx or cast and compare?

Reply:
ahh right let me add that.